src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/TestUriHelper.cs

```diff
@@ -11,39 +11,68 @@ public static class TestUriHelper
     {
         public static Uri BuildTestUri()
         {
-            return new UriBuilder("http", "localhost", GetNextPort()).Uri;
+            return BuildTestUri(null);
         }
 
         public static Uri BuildTestUri(string hint)
+        {
+            return BuildTestUri(hint, statusMessagesEnabled: false);
```

Comment:
The `TestUriHelper` class and existing `BuildTestUri()` methods are public, so they can be called directly by test classes in other assemblies, not just by `SelfHostDeployer`.  If the methods are called directly, we have no way of knowing if status messages are enabled or not, so we can't rely on them.

I made the new overload of `BuildTestUri` internal to prevent other classes from calling it directly, since it should only be used by `SelfHostDeployer`.  I will also review all existing usage of `BuildTestUri()` and `BuildTestUri(string hint)` and see if we can remove them.